ext/patches/comet/CometSearch/CometSearchManager.cpp

```diff
@@ -1837,6 +1894,90 @@ bool CometSearchManager::DoSearch()
+                  g_staticParams.inputFile.szBaseName, g_staticParams.szOutputSuffix, iFirstScan, iLastScan);
+         }
+
+         if ((fpout_mzidentml = fopen(szOutputMzIdentML, "w")) == NULL)
+         {
+            char szErrorMsg[SIZE_ERROR];
+            sprintf(szErrorMsg,  " Error - cannot write to file \"%s\".\n",  szOutputMzIdentML);
+            string strErrorMsg(szErrorMsg);
+            g_cometStatus.SetStatus(CometResult_Failed, strErrorMsg);
+            logerr(szErrorMsg);
+            bSucceeded = false;
+         }
+
+         sprintf(szOutputMzIdentMLtmp, "%s.XXXXXX", szOutputMzIdentML);
+#ifdef _WIN32
+         _mktemp_s(szOutputMzIdentMLtmp, strlen(szOutputMzIdentMLtmp) + 1);
+
+#else
+         mkstemp(szOutputMzIdentMLtmp);
+#endif
+
+         if ((fpout_mzidentmltmp = fopen(szOutputMzIdentMLtmp, "w")) == NULL)
+         {
+            char szErrorMsg[SIZE_ERROR];
+            sprintf(szErrorMsg,  " Error - cannot write to file \"%s\".\n",  szOutputMzIdentMLtmp);
+            string strErrorMsg(szErrorMsg);
+            g_cometStatus.SetStatus(CometResult_Failed, strErrorMsg);
+            logerr(szErrorMsg);
+            bSucceeded = false;
+         }
+
+         if (bSucceeded && (g_staticParams.options.iDecoySearch == 2))
+         {
+            if (iAnalysisType == AnalysisType_EntireFile)
+            {
+               sprintf(szOutputDecoyMzIdentML, "%s%s.decoy.mzid",
+                     g_staticParams.inputFile.szBaseName, g_staticParams.szOutputSuffix);
+            }
+            else
+            {
+               sprintf(szOutputDecoyMzIdentML, "%s%s.%d-%d.decoy.mzid",
+                     g_staticParams.inputFile.szBaseName, g_staticParams.szOutputSuffix, iFirstScan, iLastScan);
+            }
+
+            if ((fpoutd_mzidentml = fopen(szOutputDecoyMzIdentML, "w")) == NULL)
+            {
+               char szErrorMsg[SIZE_ERROR];
+               sprintf(szErrorMsg,  " Error - cannot write to decoy file \"%s\".\n",  szOutputDecoyMzIdentML);
+               string strErrorMsg(szErrorMsg);
+               g_cometStatus.SetStatus(CometResult_Failed, strErrorMsg);
+               logerr(szErrorMsg);
+               bSucceeded = false;
+            }
+
+            sprintf(szOutputDecoyMzIdentMLtmp, "%s.XXXXXX",szOutputDecoyMzIdentML);
+#ifdef _WIN32
+            _mktemp_s(szOutputDecoyMzIdentMLtmp, strlen(szOutputDecoyMzIdentMLtmp) + 1);
+
+#else
+            mkstemp(szOutputDecoyMzIdentMLtmp);
+#endif
+            if ((fpoutd_mzidentmltmp = fopen(szOutputDecoyMzIdentMLtmp, "w")) == NULL)
+            {
+               char szErrorMsg[SIZE_ERROR];
+               sprintf(szErrorMsg,  " Error - cannot write to decoy file \"%s\".\n",  szOutputDecoyMzIdentMLtmp);
+               string strErrorMsg(szErrorMsg);
+               g_cometStatus.SetStatus(CometResult_Failed, strErrorMsg);
+               logerr(szErrorMsg);
+               bSucceeded = false;
+            }
+

```

Comment:
Hi Yang, all the code in ext/patches actually comes from other peoples code bases. We just make small tweaks to it so it can compile as a library compatible with Crux. I don't think we want to fix small formatting issues, as they will just get clobbered the next time we update to a new version of their software.